### PR TITLE
feat: implement spec 0.7.0 changes

### PR DESCRIPTION
**Pull request by toddbaert**
* run any event handler immediately if the provider is in the associated state, not just ready
* add providerName to event details
* add STALE provider state
* update/add associated tests
* also fixed spec/test associations mismatches from previous changes

fixes: https://github.com/open-feature/java-sdk/issues/599

### CODE REVIEW

File: src/main/java/dev/openfeature/sdk/ProviderState.java [hunk 1]
```diff
@@ -4,5 +4,17 @@
...
-    READY, NOT_READY, ERROR;
+    READY, NOT_READY, ERROR, STALE;
+
+    /**
+     * Returns true if the passed ProviderEvent maps to this ProviderState.
+     * 
+     * @param event event to compare
+     * @return boolean if matches.
+     */
+    boolean matchesEvent(ProviderEvent event) {
```
Comment (Kavindu-Dodan) [inline src/main/java/dev/openfeature/sdk/ProviderState.java:15]:
```suggestion
    boolean isProviderEvent(ProviderEvent event) {
```

Note - you need to change the usage along with the renaming

Also,  we are missing `NOT_READY` state check

Comment (toddbaert) [reply src/main/java/dev/openfeature/sdk/ProviderState.java:15]: There's no "not ready" event to match.

Comment (Kavindu-Dodan) [reply src/main/java/dev/openfeature/sdk/ProviderState.java:15]: ahh yes, you are right. I was little confused here - ProviderState vs ProviderEvent

I think this naming is fine :)

File: src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java [hunk 2]
```diff
@@ -300,11 +302,11 @@ void removeHandler(String clientName, ProviderEvent event, Consumer<EventDetails
         }
     }
 
+    // here
```
Comment (Kavindu-Dodan) [inline src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java:305]: Missing code doc ?

**Review by Kavindu-Dodan — APPROVED** [review]
LGTM

Left a minor remark on a code comment, the rest looks great .! Specially thanks for checking unit tests and fixing spec references.

**Review by beeme1mr — APPROVED** [review]

File: src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java (hunk 2 above)
Comment (toddbaert) [inline src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java:305]:
```suggestion
```

File: src/main/java/dev/openfeature/sdk/EventDetails.java [hunk 3]
```diff
@@ -7,19 +7,23 @@
...
+@Data
+@SuperBuilder(toBuilder = true)
 public class EventDetails extends ProviderEventDetails {
     private String clientName;
+    private String providerName;
 
-    static EventDetails fromProviderEventDetails(ProviderEventDetails providerEventDetails) {
-        return EventDetails.fromProviderEventDetails(providerEventDetails, null);
+    static EventDetails fromProviderEventDetails(ProviderEventDetails providerEventDetails, String providerName) {
+        return EventDetails.fromProviderEventDetails(providerEventDetails, null, null);
```
Comment (thiyagu06) [inline src/main/java/dev/openfeature/sdk/EventDetails.java:17]:
```suggestion
        return EventDetails.fromProviderEventDetails(providerEventDetails, providerName, null);
```

Comment (toddbaert) [reply src/main/java/dev/openfeature/sdk/EventDetails.java]: Nice catch @thiyagu06

**Comment by beeme1mr** [comment]
Let's try and merge this tomorrow unless someone identifies a blocker.

**Comment by toddbaert** [comment]
@thiyagu06 I've accepted the change you proposed. Can you re-review?

**Review by thiyagu06 — APPROVED** [review]
LGTM 🚀

**Pull request merged**